src/components/Flow/hooks/useFlowParams.ts

```diff
@@ -6,24 +6,30 @@ export interface UseFlowParamsProps<TFlowContext> {
   requiredParams?: Array<keyof TFlowContext>
 }
 
+function hasRequiredParams<TParams extends object, TRequiredParams extends keyof TParams>(
+  params: TParams,
+  requiredParams: TRequiredParams[],
+): params is TParams & Required<Pick<TParams, TRequiredParams>> {
+  return requiredParams.every(param => param in params && typeof params[param] !== 'undefined')
+}
+
 export function useFlowParams<TFlowContext extends FlowContextInterface>({
   component,
   requiredParams = [],
 }: UseFlowParamsProps<TFlowContext>) {
   const params = useFlow<TFlowContext>()
   const { t } = useTranslation('common')
 
-  requiredParams.forEach(param => {
-    if (!(param in params)) {
-      throw new Error(
-        t('errors.missingParamsOrContext', {
-          param,
-          component,
-          provider: 'FlowProvider',
-        }),
-      )
-    }
-  })
+  if (!hasRequiredParams(params, requiredParams)) {
+    const missingParam = requiredParams.find(param => typeof params[param] === 'undefined')
+    throw new Error(
```

Comment:
What happens when the required params are missing? Do we just throw an error and break the flow? Just trying to think of cases where this could happen.